File: MonRoadtrip-back/src/main/java/monRoadtrip/model/Adresse.java
package monRoadtrip.model;

import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {

	@JsonView(JsonViews.Common.class)
	protected String numero;
	@JsonView(JsonViews.Common.class)
	protected String voie;
	@JsonView(JsonViews.Common.class)
	protected String cp;
	@JsonView(JsonViews.Common.class)
	protected String ville;
	
	public Adresse() {
	}
	
	public Adresse(String numero, String voie, String cp, String ville) {
		this.numero = numero;
		this.voie = voie;
		this.cp = cp;
		this.ville = ville;
	}


	public String getNumero() {
		return numero;
	}
	public void setNumero(String numero) {
		this.numero = numero;
	}
	public String getVoie() {
		return voie;
	}
	public void setVoie(String voie) {
		this.voie = voie;
	}
	public String getCp() {
		return cp;
	}
	public void setCp(String cp) {
		this.cp = cp;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	@Override
	public String toString() {
		return "Adresse [numero=" + numero + ", voie=" + voie + ", cp=" + cp + ", ville=" + ville + "]";
	}
	
	
	
}

File: monRoadTripAngular/src/app/model/adresse.ts
export class Adresse {
  public constructor(
    private _numero?: string | undefined,
    private _voie?: string | undefined,
    private _cp?: string | undefined,
    private _ville?: string | undefined
  ) {}

  public get numero(): string | undefined {
    return this._numero;
  }
  public set numero(value: string | undefined) {
    this._numero = value;
  }

  public get voie(): string | undefined {
    return this._voie;
  }
  public set voie(value: string | undefined) {
    this._voie = value;
  }
  public get cp(): string | undefined {
    return this._cp;
  }
  public set cp(value: string | undefined) {
    this._cp = value;
  }

  public get ville(): string | undefined {
    return this._ville;
  }
  public set ville(value: string | undefined) {
    this._ville = value;
  }
}
